sycl/doc/extensions/LocalMemory/LocalMemory.asciidoc

```diff
@@ -184,26 +161,43 @@ the group have completed execution of the kernel.
 All arguments in _args_ must be the same for all work-items in the group.
 
 `Group` must be `sycl::group`, and `T` must be trivially destructible.
-[_Note_ - These restrictions may be lifted in a future version of this
-extension.]
 
 |`template <typename T, typename Group>
  multi_ptr<T, Group::address_space> group_local_memory_for_overwrite(Group g)` |
 Constructs an object of type `T` in an address space accessible by all
 work-items in group _g_, using default initialization.  The object is
-initialized pon or before the first call to `group_local_memory`.  The storage
+initialized upon or before the first call to `group_local_memory`.  The storage
 for the object is allocated upon or before the first call to
 `group_local_memory`, and deallocated when all work-items in the group have
 completed execution of the kernel.
```

Comment:
There's a subtle difference, and the behavior is the same as it is for things like `make_unique` and `make_unique_for_overwrite`: https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Using `group_local_memory()` with an empty parameter pack is equivalent to calling `new T()`, whereas calling `group_local_memory_for_overwrite()` is equivalent to calling `new T`.  The former triggers value initialization (to zero, for integers), whereas the latter triggers default initialization (to an unknown value).

See https://godbolt.org/z/svnGjoGcj.

Reply:
It seems like this could be clarified in the description of `group_local_memory `.  Perhaps you could add a sentence like:

> Constructs an object of type T in an address space accessible by all work-items in group g, forwarding args to the constructor’s parameter list.  **If `args` is empty, the object is value initialized.**

Reply:
Sorry for the delay, and thanks for the suggested fix.  Fixed in add33183f0f08cb2ec81a33e993710e579ab8f99.